File: basic-proxy-impl/src/test/java/com/test/cases/SimpleInterfaceTest.java
package com.test.cases;

import java.util.List;
import java.util.Map;

@For(Simple.class)
public class SimpleInterfaceTest extends TestBase {

    public SimpleInterfaceTest() throws Exception {
        super();
    }
}

interface Simple<T extends Number> {
    void f0();

    int f1(int a);

    long f2(int a, long b, boolean c, List<Map<String, Object>> d, Object e);

    void f3(T t);

    static void s() {

    }

    @Override
    String toString();
}

File: basic-proxy-kotlin-impl/src/test/kotlin/com/test/cases/SimpleInterfaceTest.kt
package com.test.cases

@For(Simple::class)
class SimpleInterfaceTest : TestBase()

interface Simple<T : Number> {
    fun f0()
    fun f1(a: Int): Int
    fun f2(a: Int, b: Long, c: Boolean, d: List<Map<String, Object>>, e: Any): Long
    fun f3(t: T)

    override fun toString(): String
}
